src/sentry/integrations/api/endpoints/data_forwarding_details.py

```diff
@@ -0,0 +1,301 @@
+                project_config.is_enabled = is_enabled
+            project_config.save()
+
+    def _unenroll_removed_projects(
+        self,
+        data_forwarder: DataForwarder,
+        accessible_enrolled_projects: list[int],
+        project_ids: list[int],
+    ) -> None:
+        projects_to_unenroll = set(accessible_enrolled_projects) - set(project_ids)
+        if projects_to_unenroll:
+            DataForwarderProject.objects.filter(
+                data_forwarder=data_forwarder, project_id__in=projects_to_unenroll
+            ).delete()
+
+    @set_referrer_policy("strict-origin-when-cross-origin")
+    @method_decorator(never_cache)
+    @extend_schema(
+        operation_id="Update a Data Forwarding Configuration for an Organization",
+        parameters=[GlobalParams.ORG_ID_OR_SLUG],
+        request=DataForwarderSerializer,
+        responses={
+            200: DataForwarderModelSerializer,
+            400: RESPONSE_BAD_REQUEST,
+            403: RESPONSE_FORBIDDEN,
+        },
+    )
+    def put(
+        self, request: Request, organization: Organization, data_forwarder: DataForwarder
+    ) -> Response:
+        if request.access.has_scope("org:write"):
+            return self._update_data_forwarder_config(request, organization, data_forwarder)
+        else:  # project:write permission
+
+            project_ids, error_response = self._validate_project_ids_input(request)
+            if error_response:
+                return error_response
+
+            overrides: dict[str, Any] = request.data.get("overrides", {})
+            is_enabled: bool | None = request.data.get("is_enabled")
+
+            if not project_ids:
+                return self._handle_unenroll_all_projects(request, organization, data_forwarder)
+
+            projects = Project.objects.filter(organization_id=organization.id, id__in=project_ids)
+
+            error_response = self._validate_project_permissions(request, list(projects))
+            if error_response:
+                return error_response
+
+            error_response = self._validate_projects_exist(list(projects), project_ids)
+            if error_response:
+                return error_response
+
+            accessible_enrolled_projects = self._get_accessible_enrolled_projects(
+                request, organization, data_forwarder
+            )
+
+            self._enroll_or_update_projects(data_forwarder, project_ids, overrides, is_enabled)
+            self._unenroll_removed_projects(
+                data_forwarder, accessible_enrolled_projects, project_ids
+            )
+
+            return Response(
+                serialize(data_forwarder, request.user),
+                status=status.HTTP_200_OK,
+            )
+
+    def delete(
+        self, request: Request, organization: Organization, data_forwarder: DataForwarder
+    ) -> Response:
+        DataForwarderProject.objects.filter(data_forwarder=data_forwarder).delete()
+        data_forwarder.delete()
```

Comment:
These should be wrapped in a transaction so that if either query fail for whatever DB reasons, the entire operation fails, it doesn't delete _some_ projects (for example)